data_pipeline_api/java_api/src/main/java/uk/ramp/config/ConfigFactory.java

```diff
@@ -0,0 +1,30 @@
+package uk.ramp.config;
+
+import java.time.Instant;
+import uk.ramp.file.FileDirectoryNormaliser;
+import uk.ramp.file.FileReader;
+import uk.ramp.hash.HasherFactory;
+import uk.ramp.yaml.YamlReader;
+
+public class ConfigFactory {
+  private static final String LOCATION = "config.yaml";
+
+  public Config config(
+      YamlReader yamlReader,
+      HasherFactory hasherFactory,
+      FileReader fileReader,
+      Instant openTimestamp,
+      FileDirectoryNormaliser fileDirectoryNormaliser) {
+    var config = new ConfigReader(yamlReader, fileDirectoryNormaliser).read();
+    var failOnHashMismatch = config.failOnHashMisMatch().orElse(true);
+    var freshHash =
+        hasherFactory
+            .fileHasher(fileReader, openTimestamp)
+            .hash(fileDirectoryNormaliser.normalisePath(LOCATION));
+    var runId = config.runId().orElse(freshHash);
+    return config
+        .withFailOnHashMisMatch(failOnHashMismatch)
```

Comment:
The `failOnHashMismatch` is optional and can be left out in the config file. So a few lines above I am loading the optional value for `failOnHashMismatch` and defaulting it to false and then inserting it back into the config.

Reply:
I've removed the setting of failOnHashMismatch here now. Please see your comment about 

'With the immutable/json/yaml libraries you're using, is it possible to have a boolean with a default value of true here, rather than using an Optional <Boolean>?' 

for more info.